templates/device_name.sh

```diff
@@ -2,21 +2,18 @@
 createlvm=true
 deletelvm=false
 dsksize="0"
-disks=$(lsblk -f -d -b -n  -oNAME,SIZE,FSTYPE| egrep -v "xfs|ext3|ext4|btrfs|sr0")
-while IFS= read -r line
-do
-tmpsize=$(echo $line|awk {'print $2'})
-tmpname=$(echo $line|awk {'print $1'})
-  if [[ "$dsksize" = "0" ]]
-    then
-      dsksize=$tmpsize
-      dskname=$tmpname
-  elif [[ "$dsksize" -gt "$tmpsize" ]]
-    then
-      dsksize=$tmpsize
-      dskname=$tmpname
-  fi
-done <<< "$disks"
+
+function largest_free_disk {
+  lsblk -f -d -b -n -oNAME,SIZE | while read disk size; do
+    # ignore disks with filesystems
+    if ! lsblk -f -b -n  -oNAME,SIZE,FSTYPE -i /dev/$disk | egrep "xfs|ext3|ext4|btrfs|sr0" >/dev/null; then
+      echo -en "$disk $size"
+    fi
+  done | sort -n -k2 | head -n1 | cut -f1 -d" "
```

Comment:
This should be `sort -n -k2 -r`, so the largest disk is first (or change `head` to `tail`).

Reply:
I recall conversations about using the smallest/fastest disk for the KV DB, I don't see anything here to support that.

If we do copy this function to get the smallest disk, and the pvcreate/vgcreate/lvcreate that disk, we would need to be cautious about cases where there are only 2 disks and the smallest and largest free disk are the same disk.